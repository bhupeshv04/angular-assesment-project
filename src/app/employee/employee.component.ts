import { Component, OnInit } from '@angular/core';
import { IEmployee } from '.././Interface/employee';
import { EmployeeService } from '.././Services/employee.service';
import { ActivatedRoute, Router } from '@angular/router';

@Component({
  selector: 'app-employee',
  templateUrl: './employee.component.html',
  styleUrls: ['./employee.component.css']
})
export class EmployeeComponent implements OnInit {
 employee: IEmployee;
 statusMessage: string = 'Loading data. Please wait...';
  constructor(private _employeeService: EmployeeService,
        private _activatedRoute: ActivatedRoute,
        private _router: Router) { }

  ngOnInit() {
     let empCode: string = this._activatedRoute.snapshot.params['code'];
        this.employee=this._employeeService.getEmployeeByCode(empCode);        
    }

    onBackButtonClick() :void {
    this._router.navigate(['/employees']);
}
}