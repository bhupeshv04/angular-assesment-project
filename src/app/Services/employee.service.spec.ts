import { TestBed, async } from '@angular/core/testing';
import { EmployeeService } from '.././Services/employee.service';

describe('EmployeeServiceTest', () => {

    let employeeService;
  
      beforeEach(async(() => {

        employeeService=new EmployeeService();
      }));
  
      test('Should display a employee', () => {
        let employee=employeeService.getEmployeeByCode('emp101')
        expect(employee.code).toMatchSnapshot();
      });
  
      test('Should display a employee of code emp101 from the list', async(() => {
        let employeeList=employeeService.getEmployees();
        let emp=employeeList.filter(emp=>emp.code=='emp101')[0];
        expect(emp.code).toMatchSnapshot();
      }));
    });