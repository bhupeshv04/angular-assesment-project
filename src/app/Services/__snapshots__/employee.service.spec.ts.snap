// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeServiceTest Should display a employee 1`] = `"emp101"`;

exports[`EmployeeServiceTest Should display a employee of code emp101 from the list 1`] = `"emp101"`;
